docs/source/cicd.rst

```diff
@@ -0,0 +1,193 @@
+    GitHub user
+  * For the `infosec-dev` AWS account use the `infosec-dev-656532927350-codebuild`
+    GitHub user
+
+* Request that a GitHub Organization owner, re-deny the integration for
+  github.com/mozilla
+
+The Build Sequence
+__________________
+
+* A branch is merged into `master` in the GitHub repo
+* GitHub emits a webhook event to AWS CodeBuild indicating that a commit was
+  pushed to `master`
+* AWS CodeBuild reads the
+  `buildspec.yml <https://github.com/mozilla/MozDef/blob/master/cloudy_mozdef/buildspec.yml>`_
+  file to know what to do
+* The `install` phase of the `buildspec.yml` fetches
+  `packer <https://www.packer.io/>`_ and unzips it
+
+  * `packer` is a tool that spawns an ec2 instance, provisions it, and renders
+    an AWS Machine Image (AMI) from it.
+
+* The `build` phase of the `buildspec.yml` runs the
+  `cloudy_mozdef/ci/deploy <https://github.com/mozilla/MozDef/blob/master/cloudy_mozdef/ci/deploy>`_
+  script in the AWS CodeBuild Ubuntu 14.04 environment
+* The `deploy` script calls the
+  `build-from-cwd <https://github.com/mozilla/MozDef/blob/cfeafb77f9d4d4d8df02117a0ffca0ec9379a7d5/Makefile#L78-L79>`_
+  target of the `Makefile` which calls `docker-compose build` on the
+  `docker-compose.yml <https://github.com/mozilla/MozDef/blob/master/docker/compose/docker-compose.yml>`_
+  file, building the docker images in the AWS CodeBuild environment. These are
+  built so they can be uploaded later to DockerHub for use by developers and
+  the community.
+* The `deploy` script next calls the
+  `packer-build-github <https://github.com/mozilla/MozDef/blob/cfeafb77f9d4d4d8df02117a0ffca0ec9379a7d5/cloudy_mozdef/Makefile#L34-L36>`_
+  make target in the
+  `cloudy_mozdef/Makefile <https://github.com/mozilla/MozDef/blob/master/cloudy_mozdef/Makefile>`_
+  which calls the
+  `ci/pack_and_copy <https://github.com/mozilla/MozDef/blob/master/cloudy_mozdef/ci/pack_and_copy>`_
+  script which does the following steps
+
+  * Calls packer which launches an ec2 instance, executing a bunch of steps and
+    and producing an AMI
+  * Shares the resulting AMI with the AWS Marketplace account
+  * Copies the resulting AMI to a list of additional AWS regions
+  * Copies the tags from the original AMI to these copied AMIs in other regions
+  * Shares the AMIs in these other regions with the AWS Marketplace account
+  * Creates a blob of YAML which contains the AMI IDs. This blob will be used
+    in the CloudFormation templates
+
+* When `ci/pack_and_copy` calls packer, packer launches an ec2 instance based on
+  the configuration in
+  `cloudy_mozdef/packer/packer.json <https://github.com/mozilla/MozDef/blob/master/cloudy_mozdef/packer/packer.json>`_
+
+  * Within this ec2 instance, packer `clones the MozDef GitHub repo and checks
+    out the branch that triggered this build
+    <https://github.com/mozilla/MozDef/blob/cfeafb77f9d4d4d8df02117a0ffca0ec9379a7d5/cloudy_mozdef/packer/packer.json#L59-L60>`_
```

Comment:
Is this functionally always `master`, or could this process trigger off other builds?

Reply:
This could trigger off either `master` or off of a tagged commit, so no, not always `master`